con2fbmap/Makefile

```diff
@@ -0,0 +1,10 @@
+CC=$(PWD)/output/host/usr/bin/arm-buildroot-linux-uclibcgnueabi-gcc
```

Comment:
This doesn't look right - doesn't Buildroot set the correct $CC automatically?

Reply:
I just used the patch that Floris pointed me at. I didn't modify anything (apart from keeping it separate from the fbset pacakge) and it seemed to work.

Reply:
Eh, didn't I point you to a patch that Debian uses to add fbcon to the Debian package?!
I think it is highly unlikely Debian sets a compiler to arm-buildroot :-)

Reply:
Sorry, I obviously got confused. Mea maxima culpa! Apologies.
I'll take it out and rebuild just to make sure.

Reply:
If I take this line out, it just uses `cc` and then `strip` fails to recognise the format of the input file (cos I think it would have built it for x86).
So I don't know why it's not picking up the buildroot macros. Any ideas?

Reply:
In the buildroot package file try if it works better with just:

```
define CON2FBMAP_BUILD_CMDS
	$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)
endef
```

Might also want to simplify the Makefile to say:

```
con2fbmap: con2fbmap.c
	$(CC) con2fbmap.c -o con2fbmap
```

- First target in Makefile is default target. Can leave out the "all" stuff.
- Don't think it is necessary to compile statically.

Reply:
Something else must be wrong, maybe in con2fbmap.mk?
1, If I leave out the all: target, it complains and won't build at all.
2. Even with the above changes, it still just uses `cc`, so fails.

Reply:
1) Make sure you do the change to the buildroot .mk first.
You had "$(MAKE) -C $(@D) all" first, and that does mention "all" explicitly.

If you had "$(MAKE) -C $(@D)" without "all", it would use default target.

That will still not set CC and LD for you though.
That is what the $(TARGET_CONFIGURE_OPTS) does set, so hence my recommended change:

```   
$(MAKE) $(TARGET_CONFIGURE_OPTS) -C $(@D)
```

That is just one way to Rome though.
Could also use:

```
$(MAKE) CC="$(TARGET_CC)" LD="$(TARGET_LD)" -C $(@D)
```

Would give the same result.
TARGET_CONFIGURE_OPTS sets CC and LD and a bunch options more.
See for list: https://github.com/raspberrypi/noobs/blob/18db2628678e54d1a431fe6f21dfb9e19a962425/buildroot/package/Makefile.in#L249

Reply:
Doh! OK, I see where I went wrong. Many thanks for your patience.